src/hotspot/cpu/aarch64/gc/shenandoah/shenandoahBarrierSetAssembler_aarch64.cpp

```diff
@@ -390,15 +390,11 @@ void ShenandoahBarrierSetAssembler::store_check(MacroAssembler* masm, Register o
 
   if (UseCondCardMark) {
     Label L_already_dirty;
-    __ membar(Assembler::StoreLoad);
```

Comment:
I believe this StoreStore barrier is for preventing race against concurrent refinement. Question is that, generational Shenandoah going to implement concurrent refinement in the future?

Reply:
We plan to start work on concurrent remembered set scanning very soon. I'm not far enough into it yet to know if we'll need these barriers, but it looks like we might.

Reply:
Either way, this thing is gone with https://bugs.openjdk.java.net/browse/JDK-8261309 -- and it was needed for CMS. I would prefer to drop it to match upstream, and then reintroduce the barriers iff they are needed.

Reply:
I agree, thanks for the link and the background on the change.